File: Neetcode/ArrsHashes/isanagram.py
# https://neetcode.io/problems/is-anagram
# https://leetcode.com/problems/valid-anagram/submissions/1072995158/


class Solution(object):
    def isAnagram(self, s, t):
        track = {}
        for l in s:
            if l not in track:
                track[l] = 1
            else:
                track[l] += 1
        for l in t:
            if l not in track:
                return False
            else:
                track[l] -= 1
        for val in track.values():
            if val:
                return False
        return True

File: Neetcode/ArrsHashes/isanagram.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
 public:
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) return false;

    unordered_map<char, int> a;
    unordered_map<char, int> b;

    for (int i = 0; i < s.size(); i++) {
      a[s[i]]++;
      b[t[i]]++;
    }

    return a == b;
  }
};